resources/data-provider.mjs

```
@@ -0,0 +1,149 @@
+        } catch (error) {
+            return false;
+        }
+    }
+
+    _freezeSuites() {
+        Object.freeze(this._suites);
+        this._suites.forEach((suite) => {
+            if (!suite.tags)
+                suite.tags = [];
+            if (suite.url.startsWith("experimental/"))
+                suite.tags.unshift("all", "experimental");
+            else
+                suite.tags.unshift("all");
+            suite.enabled = suite.tags.includes("default");
+            Object.freeze(suite.tags);
+            Object.freeze(suite.steps);
+        });
+    }
+
+    _freezeTags() {
+        Object.freeze(this._tags);
+    }
+
+    async init() {
+        if (params.config) {
+            try {
+                const configUrl = new URL(params.config);
+                // Don't fetch if the URL is from DISALLOWED_DOMAINS
+                if (DISALLOWED_DOMAINS.includes(configUrl.hostname)) {
+                    console.warn(`Configuration fetch from a disallowed domain: ${params.config}. Loading default suites.`);
+                    this._loadDefaultSuites();
+                    return;
+                }
+
+                const response = await fetch(params.config);
+                // ✅ Validate that the network request was successful
+                if (!response.ok)
+                    throw new Error(`Could not fetch config: ${response.status}`);
+
+                const config = await response.json();
+                // ✅ Validate the structure of the fetched config object
+                if (!config || !Array.isArray(config.suites))
+                    throw new Error("Could not find a valid config structure!");
+
+                config.suites.flatMap((suite) => suite.tags || []).forEach((tag) => this._tags.add(tag));
+                config.suites.forEach((suite) => {
+                    // ✅ Validate each suite object before processing
+                    if (suite && suite.url && this._isValidUrl(suite.url))
```

Why doesn't this result in an exception instead?
Silently skipping a suite when its URL is invalid sounds rather misleading.

---

throwing an error now